sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -162,22 +162,38 @@ case class HiveTableScanExec(
     }
   }
 
-  // exposed for tests
-  @transient lazy val rawPartitions = {
```

Comment:
to reduce code diff, I think we can keep it, but exclude `relation.prunedPartitions`. Then `prunedPartitions` can be defined as
```
if (relation.prunedPartitions.nonEmpty) {
  ...
} else {
  rawPartitions
}
```

Reply:
> else {
>   rawPartitions
> }

sure, but rawPartitions need to be pruned if HIVE_METASTORE_PARTITION_PRUNING disabled.

Already updated, please help review, thanks.